src/components/profile/Profile.ts

```diff
@@ -7,62 +7,120 @@
       return html`
         <div class="contact-item">
           <md-badge circle color="violet">
             <md-icon name="icon-email-active_12" size="8"></md-icon>
           </md-badge>
-          <span>${this.presetTags?.email}</span>
+          <span>${this.contactData?.email}</span>
         </div>
       `;
     }
 
+    dataPointFilter(dataPoint: string) {
+      // Usage agnostic, simply retrieves the usable data. Specific to CJaaS API
+      const dataAttribute = this.profileData?.filter((x: any) => x.query.Metadata === dataPoint);
+      return dataAttribute[0]?.result[0] ? dataAttribute[0].result[0] : undefined;
+    }
+
+    extractDataPoints(update?: boolean) {
+      if ((!this.contactData && this.profileData) || update) {
+        const contactDetails = {
+          name: this.dataPointFilter("firstName"),
+          email: this.dataPointFilter("email"),
+          label: this.dataPointFilter("label"),
+          imgSrc: this.dataPointFilter("imgSrc")
+        };
+        this.contactData = contactDetails;
+      }
+    }
+
     getTopContent() {
-      const name = this.presetTags?.name?.join(" ");
+      const name = this.contactData?.name || "";
       return html`
         <section class="top-content">
-          <md-avatar .title="${name}" alt=${name} src=${this.profile?.picture || undefined} .size=${48}></md-avatar>
-          <h5 title="Name" class="customer-name">
-            ${name}
-          </h5>
-          <h5 title="Label" class="customer-label">
-            ${this.presetTags?.label ? this.presetTags?.label?.join(" ") : "VIP Customer"}
-          </h5>
-          ${this.contactItem()}
+          ${this.loading
+            ? this.getLoading()
+            : html`
+                <md-avatar
+                  .title="${name}"
+                  alt=${name}
+                  src=${ifDefined(this.contactData?.imgSrc || undefined)}
+                  .size=${48}
+                ></md-avatar>
+                <h5 title="Name" class="customer-name">
+                  ${name}
+                </h5>
+                <h5 title="Label" class="customer-label">
+                  ${this.contactData?.label}
+                </h5>
+                ${this.contactItem()}
+              `}
         </section>
       `;
     }
 
     getTable() {
-      return html`
-        <table title="Profile Details">
-          ${this.profile
-            ?.filter((x: any) => x.query.type === "table")
-            .map((x: any) => {
-              return html`
-                <tr>
-                  <td class="title">${x.query.DisplayName}</td>
-                  <td class="value">${this.getValue(x)}</td>
-                </tr>
-              `;
-            })}
-        </table>
-      `;
+      return this.loading
+        ? this.getLoading()
+        : html`
+            <table title="Profile Details">
+              ${this.profileData
+                ?.filter((x: any) => x.query.type === "table")
```

Comment:
@futurethang would it help if we did `x?.query?.type` ? or is that always guaranteed?

Reply:
Yeah those property structures are always there based on current API implementation